android/jni/com/mapswithme/maps/Framework.cpp

```diff
@@ -895,18 +895,37 @@ Java_com_mapswithme_maps_Framework_nativeGetRouteFollowingInfo(JNIEnv * env, jcl
 }
 
 JNIEXPORT jintArray JNICALL
-Java_com_mapswithme_maps_Framework_nativeGenerateRouteAltitudeChartBits(JNIEnv * env, jclass, jint width, jint height)
+Java_com_mapswithme_maps_Framework_nativeGenerateRouteAltitudeChartBits(JNIEnv * env, jclass, jint width, jint height, jobject routeAltitudeLimits)
 {
   ::Framework * fr = frm();
   ASSERT(fr, ());
 
   vector<uint8_t> imageRGBAData;
-  if (!fr->GenerateRouteAltitudeChart(width, height, imageRGBAData))
+  int32_t minRouteAltitude = 0;
+  int32_t maxRouteAltitude = 0;
+  measurement_utils::Units units = measurement_utils::Units::Metric;
+  if (!fr->GenerateRouteAltitudeChart(width, height, imageRGBAData, minRouteAltitude, maxRouteAltitude, units))
   {
     LOG(LWARNING, ("Can't generate route altitude image."));
     return nullptr;
   }
 
+  // Passing route limits.
+  jclass const routeAltitudeLimitsClass = env->GetObjectClass(routeAltitudeLimits);
+  ASSERT(routeAltitudeLimitsClass, ());
+
+  static jfieldID const minRouteAltitudeField = env->GetFieldID(routeAltitudeLimitsClass, "minRouteAltitude", "I");
```

Comment:
To @ygorshenin we doesn't worry about env, cause resulting jfieldID will be equal in any case.

To @bykoianko in fact Android doesn't unload classes, so it's valid to cache them. You can check http://android-developers.blogspot.com.by/2011/11/jni-local-reference-changes-in-ics.html for details (..."In a system without class unloading (like Android), you can treat jfieldID and jmethodID as if they were global."...)